src/printf/printf.c

```diff
@@ -78,6 +78,17 @@ extern "C" {
 # define vprintf_   vprintf
 #endif
 
+// If you have -Wfloat-equal enabled, this source code will emit a number
+// of warnings for float comparisons for equality. However, within the
+// context of their use in this library, these are ok to use. So we
+// disable the warning for this file.
+// Define PRINTF_DISABLE_WARNING_PRAGMAS to disable this block
+#ifndef PRINTF_DISABLE_WARNING_PRAGMAS
```

Comment:
Why do I want to make this conditional? Isn't conditioning on `__GNUC__` or `__clang__` enough?

Reply:
Just trying to enable flexibility - maybe there's a reason you want to compile with those warnings enabled.